src/goto-instrument/contracts/contracts.cpp

```diff
@@ -133,34 +137,66 @@ void code_contractst::check_apply_loop_contracts(
     parameter2history,
     loop_head->source_location(),
     mode,
-    havoc_code,
+    generated_code,
     ID_loop_entry);
 
   // Create 'loop_entry' history variables
-  insert_before_swap_and_advance(goto_function.body, loop_head, havoc_code);
+  insert_before_swap_and_advance(goto_function.body, loop_head, generated_code);
 
   // Generate: assert(invariant) just before the loop
   // We use a block scope to create a temporary assertion,
   // and immediately convert it to goto instructions.
   {
     code_assertt assertion{invariant_expr()};
     assertion.add_source_location() = loop_head->source_location();
-    converter.goto_convert(assertion, havoc_code, mode);
-    havoc_code.instructions.back().source_location_nonconst().set_comment(
+    converter.goto_convert(assertion, generated_code, mode);
+    generated_code.instructions.back().source_location_nonconst().set_comment(
       "Check loop invariant before entry");
   }
 
   // havoc the variables that may be modified
-  havoc_if_validt havoc_gen(modifies, ns);
-  havoc_gen.append_full_havoc_code(loop_head->source_location(), havoc_code);
+  modifiest modifies;
```

Comment:
Is "modifiest" a good choice when we are talking about assigns?

Reply:
This is a good point. Our assigns clauses don't have _modifies_ semantics. I'm going to make a separate GitHub issue to track this, because it requires changes outsides of contracts code (`loop_utils.{h,cpp}` and also `function_modifies.{h,cpp})`.

Reply:
Created #6467.